interoperability-tests/Testing-between-Gramine-and-IntelSGXSDK-ratls.md

```diff
@@ -0,0 +1,158 @@
+
+Gramine provides an example program [ra-tls-mbedtls](https://github.com/gramineproject/gramine/tree/master/CI-Examples/ra-tls-mbedtls) based on mbedtls for RA-TLS. We use the client program in it for testing.
+
+1. Download and checkout correct Gramine version
+
+```sh
+git clone https://github.com/gramineproject/gramine.git
+cd gramine/
+git checkout dimakuv/interoperable-ra-tls
+```
+
+2. Build and install Gramine system-wide (in debug mode, so it can be GDBed if needed)
+
+```sh
+meson setup build-debug/ --buildtype=debug -Dskeleton=enabled -Ddirect=enabled -Dsgx=enabled -Ddcap=enabled -Dsgx_driver=upstream -Dsgx_driver_include_path=/usr/include/x86_64-linux-gnu/
+
+ninja -C build-debug/
+sudo ninja -C build-debug/ install
+```
+
+3. build and start the RA-TLS client
+
+```sh
+cd CI-Examples/ra-tls-mbedtls/
+make client
+RA_TLS_ALLOW_DEBUG_ENCLAVE_INSECURE=1 RA_TLS_ALLOW_OUTDATED_TCB_INSECURE=1  ./client dcap
+```
+
+> Use `RA_TLS_ALLOW_DEBUG_ENCLAVE_INSECURE=1` to allow debug enclaves, and use `RA_TLS_ALLOW_OUTDATED_TCB_INSECURE=1` to allow outdated TCB in quote.
+
+
+# Gramine as attester and Intel SGXSDK ra-tls as verifier
+
+## Gramine
+
+We use the example program ra-tls-mbedtls provided by Gramine to start an RA-TLS server.
+
+1. Download and checkout correct Gramine version
+
+```sh
+git clone https://github.com/gramineproject/gramine.git
+cd gramine/
+git checkout dimakuv/interoperable-ra-tls
+```
+
+2. Build and install Gramine system-wide (in debug mode, so it can be GDBed if needed)
+
+```sh
+meson setup build-debug/ --buildtype=debug -Dskeleton=enabled -Ddirect=enabled -Dsgx=enabled -Ddcap=enabled -Dsgx_driver=upstream
+ninja -C build-debug/
+sudo ninja -C build-debug/ install
+```
+
+3. build and start the RA-TLS server
+
+```sh
+cd CI-Examples/ra-tls-mbedtls/
+make app dcap RA_TYPE=dcap
+gramine-sgx ./server
+```
+
+Now the Gramine server listens on port 4433, clients can connect to it and examine its cert.
+
+Here is a certificate generated by Gramine: [gramine-cert.pem](./gramine-cert.pem)
+
+## Intel SGXSDK ra-tls
+
+There are 2 clients provided in Intel SGXSDK ra-tls SampleAttestedTLS, [ra-tls-sample](https://github.com/intel/linux-sgx/tree/master/SampleCode/SampleAttestedTLS), client in trusted(SampleAttestedTLS/client) and client in untrusted environment(SampleAttestTLS/non_enc_client), both could be used to test RA-TLS connections. 
+
+> Note that the clients in Intel SGXSDK SampleAttestedTLS can also be run as a verifier.
+
+1. Get source code of SampleAttestedTLS
+
+```sh
+git clone https://github.com/intel/linux-sgx/tree/master/SampleCode/SampleAttestedTLS
+```
+
+2. Build the Sample
+
+```sh
+cd SampleAttestedTLS
+
+make
+```
+
+3. Start client
+In SampleAttestedTLS, we provide 2 types of clients, one run inside enclave, another run in untrusted side: 
+
+Client run inside enclave
+```sh
+./client/host/tls_client_host ./client/enc/tls_client_enclave.signed.so -server:localhost -port:4433
+```
+
+Non-enclave client
```

Comment:
Please make it a bullet list, and indent it such that it renders under the (3) item.